src/components/organisms/UserProfileModal/UserProfileModal.tsx

```diff
@@ -141,25 +139,26 @@ const SuspectedLocation: React.FC<{
   currentVenue: WithId<AnyVenue>;
 }> = ({ user, currentVenue }) => {
   // @debt This will currently load all venues in firebase into memory.. not very efficient
-  useFirestoreConnect("venues");
-  const allVenues = useSelector(orderedVenuesSelector);
+  const { relatedVenues } = useRelatedVenues({
+    currentVenueId: currentVenue.id,
+  });
 
   const suspectedLocation = useMemo(
     () => ({
-      venue: allVenues?.find(
+      venue: relatedVenues?.find(
```

Comment:
`relatedVenues` is guaranteed to always be an array:
```suggestion
      venue: relatedVenues.find(
```

Reply:
done 1e8fb879bcd345ad7c00ceb417ac1dfa78c5bbff